eclair-core/src/main/scala/fr/acinq/eclair/payment/Bolt12Invoice.scala

```diff
@@ -203,7 +203,7 @@ object MinimalBolt12Invoice {
       OfferTypes.InvoiceCreatedAt(createdAt),
       OfferTypes.InvoicePaymentHash(paymentHash),
       OfferTypes.InvoiceAmount(amount),
-      OfferTypes.InvoiceNodeId(offer.nodeId),
+      OfferTypes.InvoiceNodeId(offer.contactInfos.head.nodeId),
```

Comment:
This won't work in all cases, will it? Do we need to change the constructor to ensure we use the `nodeId` from the blinded path that was used?

Reply:
`MinimalBolt12Invoice` is not a real invoice, the goal is to keep track of the important parts of the real invoice without having to put the whole invoice in the `OfferPaymentMetadata`.
Setting the correct nodeId here would require storing it in the `OfferPaymentMetadata`.
Is it really important to know which of the paths the payer used to retrieve the invoice?

Reply:
You're right, it doesn't matter in the `MinimalBolt12Invoice`, it's not worth the additional complexity!